File: src/types/template.ts
export interface TemplateField {
  key: string;
  label: string;
  type: string;
  default?: string;
  min?: number;
  max?: number;
  accept?: string[];
}

export interface TemplateInfo {
  id: string;
  name: string;
  path: string;
  fields: TemplateField[];
}

export interface Manifest {
  templates: TemplateInfo[];
}

File: src-tauri/src/domain/template.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct OverlayField {
    pub key: String,
    pub label: String,
    #[serde(rename = "type")]
    pub field_type: String,
    #[serde(default)]
    pub default: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub min: Option<f64>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub max: Option<f64>,
}

#[derive(Debug, Clone, Serialize)]
pub struct TemplateInfo {
    pub id: String,
    pub name: String,
    pub path: String,
    pub fields: Vec<OverlayField>,
}

#[derive(Debug, Clone, Serialize)]
pub struct Manifest {
    pub templates: Vec<TemplateInfo>,
}
